### ISSUE TITLE
Add attribute resolution logic

### ISSUE THREAD

**Issue by nfx** [databrickslabs/ucx#1864]
Can you also add few more tests:

```
x = dbutils.notebook 
x.run("foo")
```
And 
```
y = dbutils
y.notebook.run("foo")
```
And
```
z = dbutils.notebook.run
z("foo")
```
And False positive test: 
```
x = Some()
x.run("foo")
```
_Originally posted by @nfx in https://github.com/databrickslabs/ucx/pull/1860#discussion_r1631503803_

**Comment by ericvergnaud** [comment]
Partially implemented by #2114

**Comment by nfx** [comment]
done in https://github.com/databrickslabs/ucx/pull/2114
